### PR TITLE
chore(.github): add release cd

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
TODO:

- [x] feat(base): install nvm, node, pnpm
- [x] chore(.github): add release action
- [x] ensure conventional commit
- [x] increment tag based on conventional commit

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
_and, of course, when I can open it as a draft, I forget to_

**Comment by ShaunSHamilton** [comment]
@raisedadead This should be ready.

**TODO:**

- [ ] Create at least 1 manual release
- [ ] Create READ_ONLY github token, and add to secret
- [ ] Add `DOCKERHUB_USERNAME` to secrets (GHCR username)
- [ ] Add `DOCKERHUB_TOKEN` to secrets (GHCR password)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          node-version: 18
+      - run: |
+          npm i
+          npx commitlint --from HEAD~1 --to HEAD --verbose
+
+  build:
+    needs: commitlint
+    runs-on: ubuntu-22.04
+    if: ${{ contains(fromJSON('["!:", "BREAKING CHANGE:", "patch", "minor", "major"]'), github.event.head_commit.message) || !(startsWith('chore', github.event.head_commit.message) || startsWith('refactor', github.event.head_commit.message) || startsWith('docs', github.event.head_commit.message)) }}
```
Comment (raisedadead) [inline .github/workflows/release.yml:27]: This is a complex condition, do you mind explaining in a comment what this does for future reference?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          registry: ghcr.io
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Tag Version
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm i
```
Comment (raisedadead) [inline .github/workflows/release.yml:44]: Any reason to go with npm, because we are favoring pnpm now.

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (ShaunSHamilton) [reply .github/workflows/release.yml:44]: It is an extra thing to install, and this is such a simple repo - no workspaces with complex deps.

File: .github/workflows/release.yml (hunk 1 above)
Comment (ShaunSHamilton) [reply .github/workflows/release.yml:27]:
```suggestion
    # Only run the release job if the commit message suggests a release:
    # It is assumed commits starting with "chore" | "refactor" | "docs" do not warrant a release
    if: ${{ contains(fromJSON('["!:", "BREAKING CHANGE:", "patch", "minor", "major"]'), github.event.head_commit.message) || !(startsWith('chore', github.event.head_commit.message) || startsWith('refactor', github.event.head_commit.message) || startsWith('docs', github.event.head_commit.message)) }}
```

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
OK - go ahead and try it out.

**Pull request merged**